Tools/patchcheck/patchcheck.py

```diff
@@ -260,43 +175,15 @@ def regenerated_pyconfig_h_in(file_paths):
     else:
         return "not needed"
 
-def ci(pull_request):
-    if pull_request == 'false':
-        print('Not a pull request; skipping')
-        return
-    base_branch = get_base_branch()
-    file_paths = changed_files(base_branch)
-    python_files = [fn for fn in file_paths if fn.endswith('.py')]
-    c_files = [fn for fn in file_paths if fn.endswith(('.c', '.h'))]
-    doc_files = [fn for fn in file_paths if fn.startswith('Doc') and
-                 fn.endswith(('.rst', '.inc'))]
-    fixed = []
-    fixed.extend(normalize_whitespace(python_files))
-    fixed.extend(normalize_c_whitespace(c_files))
-    fixed.extend(normalize_docs_whitespace(doc_files))
-    if not fixed:
-        print('No whitespace issues found')
-    else:
-        print(f'Please fix the {len(fixed)} file(s) with whitespace issues')
-        print('(on UNIX you can run `make patchcheck` to make the fixes)')
-        sys.exit(1)
 
 def main():
     base_branch = get_base_branch()
     file_paths = changed_files(base_branch)
-    python_files = [fn for fn in file_paths if fn.endswith('.py')]
-    c_files = [fn for fn in file_paths if fn.endswith(('.c', '.h'))]
-    doc_files = [fn for fn in file_paths if fn.startswith('Doc') and
-                 fn.endswith(('.rst', '.inc'))]
+    has_doc_files = any(fn for fn in file_paths if fn.startswith('Doc') and
+                        fn.endswith(('.rst', '.inc')))
     misc_files = {p for p in file_paths if p.startswith('Misc')}
-    # PEP 8 whitespace rules enforcement.
-    normalize_whitespace(python_files)
-    # C rules enforcement.
-    normalize_c_whitespace(c_files)
-    # Doc whitespace enforcement.
-    normalize_docs_whitespace(doc_files)
```

Comment:
Sorry if I’m missing the context – aren’t these checks the original motivation for having patchcheck?
Are they replaced by something else like pre-commit which would now be a required tool in the workflow?

Reply:
There are three proposed PRS (#109854; #109890; #109891) to move the whitespace checks to pre-commit.

Hugo commented in #109854 (point 3):

> I'd like to eventually remove patchcheck.

I opened this PR as a semi-compromise; it doesn't delete patchcheck but just removes it from CI. However, if it would still be useful to keep the various ``normalize_XX`` functions in patchcheck when running locally, I've no objections to reinstating them (also cc @hugovk for his thoughts here).

Reply:
I'd like to remove patchcheck but replace it with pre-commit or similar, and want them to run in CI, and to be easily runnable locally.

I added some thoughts at https://discuss.python.org/t/do-you-use-make-patchcheck/34743/2

Reply:
Why remove a tool that exists, works and requires nothing with something else that needs to be installed?

Reply:
First of all, we've added new lint checks via existing pre-commit rules, rather than writing and maintaining them ourselves. I expect we'll add more in the future.

There's a quite a bit of overlap between the patchcheck lints and what we can also do with pre-commit, so I think it will be easier to maintain this in one place, and it will give a more consistent for users, both in the CI and locally.

We can of course keep these things in patchcheck if people find them useful there, and only add new lints to pre-commit.